src/event/metric.rs

```diff
@@ -8,6 +8,7 @@ use std::{
     fmt::{self, Display, Formatter},
     iter::FromIterator,
 };
+use structures::str::immutable::String as ImStr;
```

Comment:
Given this usage, should we standardize on a fixed name in `structures::str::imutable` instead of renaming on each import?

Reply:
I'm game. My thinking at the time was that we'd _eventually_ make `std::string::String` the unusual stringy type if this work paid off, which is why I staked for `String`. Having `ImStr` everywhere is pretty awkward so I'm flexible here. I do want to avoid names that suggest at the implementation of the type, so `BoxStr` leaves me feeling a little cold. Do we need to adjust the tree again if we introduce an `InterenedStr` or similar? But, I can also accept that we'd want to leave the implementation at least hinted at to give people some criteria to choose between types.

Reply:
I agree on avoiding `BoxStr`, but have no other strong opinions on the name. I just think we should avoid a setup that effectively *requires* us to rename on `use`.

Reply:
You're right, it doesn't sit well. How does `ImmStr` strike your ear?

Reply:
I have no strong objection. @lukesteensen?